### ISSUE TITLE
AttributeError: 'str' object has no attribute 'get'

### ISSUE THREAD

**Issue by maxxxter** [DIGITALCRIMINAL/UltimaScraper#1900]
Type: Profile
Type: Stories
Traceback (most recent call last):
  File "D:\OnlyFans-7.6.1\start_ofd.py", line 104, in <module>
    asyncio.run(main())
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1520.0_x64__qbz5n2kfra8p0\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1520.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 646, in run_until_complete
    return future.result()
  File "D:\OnlyFans-7.6.1\start_ofd.py", line 86, in main
    api = await main_datascraper.start_datascraper(
  File "D:\OnlyFans-7.6.1\datascraper\main_datascraper.py", line 94, in start_datascraper
    await main_helper.process_names(
  File "D:\OnlyFans-7.6.1\helpers\main_helper.py", line 966, in process_names
    result = await module.start_datascraper(authed, username, site_name)
  File "D:\OnlyFans-7.6.1\modules\onlyfans.py", line 161, in start_datascraper
    await prepare_scraper(authed, site_name, item)
  File "D:\OnlyFans-7.6.1\modules\onlyfans.py", line 784, in prepare_scraper
    highlights = await subscription.get_highlights()
  File "D:\OnlyFans-7.6.1\apis\onlyfans\classes\create_user.py", line 270, in get_highlights
    results = [create_highlight(x) for x in results]
  File "D:\OnlyFans-7.6.1\apis\onlyfans\classes\create_user.py", line 270, in <listcomp>
    results = [create_highlight(x) for x in results]
  File "D:\OnlyFans-7.6.1\apis\onlyfans\classes\create_highlight.py", line 3, in __init__
    self.id: int = option.get("id")
AttributeError: 'str' object has no attribute 'get'

tried updating as suggested in another post, didn't help

tqdm Version: 4.64.0

**Comment by maxxxter** [comment]
worked great as recently as a few days ago, nothing on my end has changed that I know of.

**Comment by requader** [comment]
> worked great as recently as a few days ago, nothing on my end has changed that I know of.

Same here. What's weird is that I have 7.6.1, which isn't in the list of releases (most recent is 7.4.1). Not sure if I'm missing something here but doesn't work in Docker or local either way.

**Comment by whats-happening-rightnow** [comment]
bump, seeing the same thing.

one of the api calls is returning a 400 now:

![image](https://user-images.githubusercontent.com/12958294/181069678-a958f227-6407-4d85-bbaa-94d79e69f9cf.png)

**Comment by dmingledorff** [comment]
Yep getting the same error.  Was working fine several days ago.

**Comment by ewhgae457yaefgae4** [comment]
This update below should fix the issue. It was caused by onlyfans changing part of their website.
anonfiles.com/G1lcNf06yf/OnlyfansDL_Devbuild_exe

**Comment by dmingledorff** [comment]
> This update below should fix the issue. It was caused by onlyfans changing part of their website. anonfiles.com/G1lcNf06yf/OnlyfansDL_Devbuild_exe

Why is that an executable?

**Comment by whats-happening-rightnow** [comment]
> > This update below should fix the issue. It was caused by onlyfans changing part of their website. anonfiles.com/G1lcNf06yf/OnlyfansDL_Devbuild_exe
> 
> Why is that an executable?

reported @ewhgae457yaefgae4 for dodgy behavior.  ignore this fool.

**Comment by whats-happening-rightnow** [comment]
error is related to getting "highlights," so just commented out the call and it's working as usual

![image](https://user-images.githubusercontent.com/12958294/181343821-ed2b472d-ec57-4d1e-b22e-d3b6938812f7.png)

**Comment by skrrrtloder** [comment]
Hello,

I'm having the same issue. Can anyone please explain how I can fix this? I have very little experience in coding. I appreciate the help.

**Comment by maxxxter** [comment]
> error is related to getting "highlights," so just commented out the call and it's working as usual
> 
> ![image](https://user-images.githubusercontent.com/12958294/181343821-ed2b472d-ec57-4d1e-b22e-d3b6938812f7.png)

Your're awesome, Thanks!!

**Comment by Kokoraky** [comment]
> error is related to getting "highlights," so just commented out the call and it's working as usual
> 
> ![image](https://user-images.githubusercontent.com/12958294/181343821-ed2b472d-ec57-4d1e-b22e-d3b6938812f7.png)

thanks

**Comment by mbiasetti** [comment]
Thanks for the tip!

**Comment by Curiocase** [comment]
I'm having a similar issue. I'm running in docker, and i tried whats-happening-rightnows tip but still getting a similar error even after commenting out the highlights code:

`Traceback (most recent call last):
  File "/usr/src/app/./start_ofd.py", line 104, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/usr/src/app/./start_ofd.py", line 86, in main
    api = await main_datascraper.start_datascraper(
  File "/usr/src/app/datascraper/main_datascraper.py", line 94, in start_datascraper
    await main_helper.process_names(
  File "/usr/src/app/helpers/main_helper.py", line 966, in process_names
    result = await module.start_datascraper(authed, username, site_name)
  File "/usr/src/app/modules/onlyfans.py", line 161, in start_datascraper
    await prepare_scraper(authed, site_name, item)
  File "/usr/src/app/modules/onlyfans.py", line 784, in prepare_scraper
    highlights = await subscription.get_highlights()
  File "/usr/src/app/apis/onlyfans/classes/create_user.py", line 270, in get_highlights
    results = [create_highlight(x) for x in results]
  File "/usr/src/app/apis/onlyfans/classes/create_user.py", line 270, in <listcomp>
    results = [create_highlight(x) for x in results]
  File "/usr/src/app/apis/onlyfans/classes/create_highlight.py", line 3, in __init__
    self.id: int = option.get("id")
AttributeError: 'str' object has no attribute 'get'
`

**Comment by Curiocase** [comment]
disregard my comment. I'm in idiot and didn't rebuild the scraper in docker after making the changes

**Comment by anon767346** [comment]
has this been fixed in a recent commit?
@Curiocase how did you fix this with docker?

**Comment by LP560** [comment]
> has this been fixed in a recent commit? @Curiocase how did you fix this with docker?

You can just open onlyfans.py in Notepad++ and edit it then save your edit and it will work.

**Comment by Curiocase** [comment]
@anon767346 I used @whats-happening-rightnow method to fix, just with docker, it appears that another step is necessary. I'm not super savvy with this stuff, and am only basing this off of how i interpreted the situation, but running the scraper in docker, the settings from onlyfans.py were locked in from when i first set it up, so any changes in onlyfans.py after the fact wouldn't take effect(again my interpretation from a person who is NOT an expert), so for the changes to take effect, it made sense for me to rebuild the scraper in docker. 
to do so, navigate to the scrapers main folder and run in powershell:
`docker build -t only-fans .`

I did run into a separate unrelated issue, but once fixed that issue, i had no problems running the scraper as normal. I hope that helps
